### ISSUE TITLE
Extract CSSOM/DOM related utilities into a separate package

### ISSUE THREAD

**Issue by kof** [cssinjs/jss#744]
As a first step of moving to jest, we need to separate the code that needs karma and cross-browser testing, since jest doesn't support that.

We can create a small functional utility belt for CSSinJS libraries that encapsulates all the CSSOM/DOM logic and we can use the current karma setup for this library.

**Comment by kof** [comment]
Should this package be part of the monorepo? I tend to "no" for 2 reasons:
-  to make other libraries be more comfortable depending on it
- to not have karma setup with all its dependencies in this monorepo, since it would make bootstrapping longer

**Comment by HenriBeck** [comment]
Any idea for naming this package?

**Comment by kof** [comment]
cssinjs-utils is free

**Comment by kof** [comment]
mb cssom-utils is even better

**Comment by HenriBeck** [comment]
yeah cssom-utils is better

**Comment by HenriBeck** [comment]
@kof could you get started on this? That way we wouldn't need a separate test script for the babel plugin

**Comment by kof** [comment]
I want to finish babel and webpack plugins first. To kick start this we could discuss all functions that we would need to extract from JSS.

**Comment by HenriBeck** [comment]
I think the functions are mostly/all in https://github.com/cssinjs/jss/blob/master/packages/jss/src/renderers/DomRenderer.js

**Comment by kof** [comment]
Yes, but we will have to remove any JSS specific logic/knowledge from them to make them universal.

**Comment by HenriBeck** [comment]
Let's make a plan for this to move forward on switching the test setup to jest.

**Comment by kof** [comment]
I would say that someone needs to identify generic utilities needed in DOMRenderer and describe the interfaces of them. Then once its discussed move forward with implementation and releasing a package.
